### PR TITLE
doc: poll: add notes (repeated cb & cancel pending cb)

### PR DESCRIPTION
**Pull request by eladn**
Added notes to documentation of `uv_poll_t`:
- The callback will be called over-and-over again as long as the socket
  remains readable/writable.
- uv_pool_stop() cancels pending callbacks of already happened events.
  https://github.com/libuv/libuv/issues/1078

### CODE REVIEW

File: docs/src/poll.rst [hunk 1]
```diff
@@ -106,11 +106,24 @@ API
     .. note::
         Though UV_DISCONNECT can be set, it is unsupported on AIX and as such will not be set
         on the `events` field in the callback.
+        
+    .. note::
+        If one of the events UV_READABLE or UV_WRITABLE are set, the callback will be called over-and-over again, as long as the given fd/socket remains readable or writable accordingly. Particullary in each of the following scenerios:
+        
+        * The callback has been called because the socket became readable/writable and the callback did not conduct a read/write on this socket at all.
+        * The callback commited a read on the socket, and have not read all the available data (when UV_READABLE is set).
```
Comment (txdv) [inline docs/src/poll.rst]: shouldn't it be "and has not read all"?

File: docs/src/poll.rst [hunk 2]
```diff
@@ -106,11 +106,24 @@ API
...
+        In all of the above listed scenerios, the socket remains readable or writable and hence the callback will be called again (depending on the events set in the bitmask).
 
     .. versionchanged:: 1.9.0 Added the UV_DISCONNECT event.
 
 .. c:function:: int uv_poll_stop(uv_poll_t* poll)
 
     Stop polling the file descriptor, the callback will no longer be called.
+    
+    .. note::
+        Calling :c:func:`uv_poll_stop` on a handle that its socket state has already been changed, but the callback has not called yet, would also cancel the pending callback.
```
Comment (txdv) [inline docs/src/poll.rst]: "on handle that its" sounds incorrect to me, grammar wise.

File: docs/src/poll.rst [hunk 3]
```diff
@@ -106,11 +106,24 @@ API
     .. note::
         Though UV_DISCONNECT can be set, it is unsupported on AIX and as such will not be set
         on the `events` field in the callback.
+        
+    .. note::
+        If one of the events UV_READABLE or UV_WRITABLE are set, the callback will be called over-and-over again, as long as the given fd/socket remains readable or writable accordingly. Particullary in each of the following scenerios:
```
Comment (saghul) [inline docs/src/poll.rst]: put the constants between backticks

File: docs/src/poll.rst [hunk 4]
```diff
@@ -106,11 +106,24 @@ API
...
+        
+        * The callback has been called because the socket became readable/writable and the callback did not conduct a read/write on this socket at all.
+        * The callback commited a read on the socket, and have not read all the available data (when UV_READABLE is set).
+        * The callback commited a write on the socket, but it remained writable afterwards (when UV_WRITABLE is set).
+        * The socket has already became readable/writable before calling :c:func:`uv_poll_start` on a poll handle associated with this socket, and since then the state of the socket did not changed.
+        
+        In all of the above listed scenerios, the socket remains readable or writable and hence the callback will be called again (depending on the events set in the bitmask).
```
Comment (saghul) [inline docs/src/poll.rst]: maybe add some "this is known as level triggering".

**Comment by cjihrig** [comment]
Can you wrap lines at 80 characters please.

File: docs/src/poll.rst (hunk 3 above)
Comment (cjihrig) [inline docs/src/poll.rst]: Typo: particularly

**Comment by saghul** [comment]
@eladn ping

File: docs/src/poll.rst (hunk 3 above)
Comment (eladn) [inline docs/src/poll.rst]: hi! those constants are refrenced all over this file without backticks. should we put them all between backticks?

File: docs/src/poll.rst (hunk 1 above)
Comment (eladn) [inline docs/src/poll.rst]: right! thanks. fixed.

File: docs/src/poll.rst (hunk 2 above)
Comment (eladn) [inline docs/src/poll.rst]: Hi! i agree. may you please suggest another phrasing to this sentence?

File: docs/src/poll.rst (hunk 3 above)
Comment (eladn) [inline docs/src/poll.rst]: Thanks! fixed.

File: docs/src/poll.rst (hunk 4 above)
Comment (eladn) [inline docs/src/poll.rst]: cool. added.

**Comment by eladn** [comment]
Hi. new commit has been pushed relating the issues described in the above comments. https://github.com/libuv/libuv/pull/1100/commits/d8a809a123346cd3c763e9b1aabfe87e874f1b2e
However, 2 issues have not been fixed yet:
1. @saghul : I replied to your comment regarding the backticks around constants.
2. @txdv : I couldn't rephrase the sentence in a different manner. may you suggest another phrasing?

**Comment by saghul** [comment]
@eladn Please add backticks only to your changes, not the entire file, that would add too much noise to the PR...

File: docs/src/poll.rst [hunk 5]
```diff
@@ -106,11 +106,24 @@ API
...
+        In all of the above listed scenerios, the socket remains readable or writable and hence the callback will be called again (depending on the events set in the bitmask). This behaviour is known as level triggering.
 
     .. versionchanged:: 1.9.0 Added the UV_DISCONNECT event.
 
 .. c:function:: int uv_poll_stop(uv_poll_t* poll)
 
     Stop polling the file descriptor, the callback will no longer be called.
+    
+    .. note::
+        Calling :c:func:`uv_poll_stop` on a handle that its socket state has already been changed, but the callback has not called yet, would also cancel the pending callback.
```
Comment (eladn) [inline docs/src/poll.rst]: done.

File: docs/src/poll.rst [hunk 6]
```diff
@@ -106,11 +106,24 @@ API
...
+        
+    .. note::
+        If one of the events UV_READABLE or UV_WRITABLE are set, the callback will be called again, as long as the given fd/socket remains readable or writable accordingly. Particularly in each of the following scenerios:
+        
+        * The callback has been called because the socket became readable/writable and the callback did not conduct a read/write on this socket at all.
+        * The callback commited a read on the socket, and has not read all the available data (when UV_READABLE is set).
+        * The callback commited a write on the socket, but it remained writable afterwards (when UV_WRITABLE is set).
+        * The socket has already became readable/writable before calling :c:func:`uv_poll_start` on a poll handle associated with this socket, and since then the state of the socket did not changed.
+        
+        In all of the above listed scenerios, the socket remains readable or writable and hence the callback will be called again (depending on the events set in the bitmask). This behaviour is known as level triggering.
```
Comment (eladn) [inline docs/src/poll.rst]: done.

File: docs/src/poll.rst [hunk 7]
```diff
@@ -106,11 +106,24 @@ API
     .. note::
         Though UV_DISCONNECT can be set, it is unsupported on AIX and as such will not be set
         on the `events` field in the callback.
+        
+    .. note::
+        If one of the events UV_READABLE or UV_WRITABLE are set, the callback will be called again, as long as the given fd/socket remains readable or writable accordingly. Particularly in each of the following scenerios:
+        
+        * The callback has been called because the socket became readable/writable and the callback did not conduct a read/write on this socket at all.
+        * The callback commited a read on the socket, and has not read all the available data (when UV_READABLE is set).
+        * The callback commited a write on the socket, but it remained writable afterwards (when UV_WRITABLE is set).
```
Comment (eladn) [inline docs/src/poll.rst]: done.

**Comment by eladn** [comment]
@cjihrig : i cannot break the line inside of `note` section when using bullets. it breaks the bullets indentation when rendering to html.

**Comment by cjihrig** [comment]
@santigimeno @saghul is there anything still needed on this one besides breaking the long lines?

**Comment by vtjnash** [comment]
Seems like good notes to add (assuming we are committed to level-triggering there, which seems certainly true at least for all of v1).

> it breaks the bullets indentation when rendering to html.

I think it should be fine. Just indent the line to match the previous: http://docutils.sourceforge.net/docs/user/rst/quickref.html#bullet-lists

**Review by santigimeno — APPROVED** [review]

**Comment by santigimeno** [comment]
LGTM and sorry for the infinite delay.

**Review by cjihrig — APPROVED** [review]

**Pull request merged**